yosys-synlig: drop

gonsolo opened the pull request:
Synlig has stopped development, that last commit is from Dec 2024, it seems yosys-slang is used nowadays.

GaetanLepage reviewed (changes requested):
Is this package actually broken or simply "unmaintained"?
If it still builds, marking is as `broken` is not the right action.

Also, is there an official statement from upstream stating that development has stopped? Or is your observation just based on the date of the latest commit?

If it is indeed "archived/unmaintained" AND if it does not make any sense to keep it around, then we should properly drop the package. This means removing the files and adding an entry in `pkgs/top-level/aliases.nix`.

gonsolo committed:
synlig: drop

The upstream project is no longer maintained and is not compatible with
newer yosys versions.

gonsolo commented:
> If @hzeller agrees, then sure, let's drop it.

I tried several hours to get it to work but the API breakage is just too big.
If someone really needs it he can stick to an older nix version or fix upstream, then this package can be easily resurrected.
To make it easy for @hzeller I pushed the necessary changes to drop it.

NickCao commented:
Correct commit message would be: `yosys-synlig: drop`, not a big deal though.

gonsolo commented:
> Backport failed for `release-25.11`, because it was unable to cherry-pick the commit(s).

This is a package removal; backporting to stable is not necessary as it would be a breaking change for 25.11 users.
